### PR TITLE
Pass orderId to DCM u8 parameter

### PR DESCRIPTION
**Pull request by MicBosi**
#### Changes proposed in this Pull Request

* Upon purchase we now pass the transaction id to Floodlight via the `u8` param.

Note:` u6` and `u7` prarams after further inspection did not need any fixing.

#### Testing instructions

Visit: https://calypso.live/?branch=fix/pass_transaction_id_to_floodlight_u8

Enable tracking and logging via JS console:
```
document.cookie = 'flags=gdpr-banner,google-analytics,ad-tracking; path=/; domain=.' + document.location.hostname.split( '.' ).slice( -2 ).join( '.' );
localStorage.setItem( 'debug', 'calypso:analytics:*' );
```

Reload browser ♻️ 

The JS console should now show various lines with: (filter by `isAdTrackingAllowed`)

```
calypso:analytics:utils isAdTrackingAllowed: true
```

Purchase a plan for a test website using free credits.

In your JS log you should see the `u8` parameter being passed in the dataLayer:

image

and reflected in the Network activity:

image

### CODE REVIEW

**Comment by southp** [comment]
@MicBosi Sorry for my ignorance here. Could you help me understand why turning off Icon Media and AdRoll supports is part of this? Just asking because it would be easier to test if we could separate the order id update and the vendor update out, if possible :)

**Comment by MicBosi** [comment]
Thansk @southp for stopping by, the PR is still WIP, I need to put proper instructions etc., I'll split the changes up for ya, no problem 👍

**Comment by MicBosi** [comment]
@southp I've updated the testing instructions, this is now ready for review, thanks!

File: client/lib/analytics/ad-tracking.js [hunk 1]
```diff
@@ -587,13 +587,9 @@ export function trackCustomAdWordsRemarketingEvent( properties ) {
...
-	// Note: reduce() requires a non 0-length array.
-	const jetpackCost =
-		0 === cart.products.length
-			? 0
-			: cart.products
-					.map( product => ( productsValues.isJetpackPlan( product ) ? product.cost : 0 ) )
-					.reduce( ( accumulator, cost ) => accumulator + cost );
+	const jetpackCost = cart.products
+		.map( product => ( productsValues.isJetpackPlan( product ) ? product.cost : 0 ) )
+		.reduce( ( accumulator, cost ) => accumulator + cost, 0 );
```
Comment (taggon) [inline client/lib/analytics/ad-tracking.js]: I think we don't need `.map` here.

```js
jetpackCost = cart.products.reduce( ( sum, product ) => sum + ( productValues.isJetpackPlan( product ) ? product.cost : 0 ), 0 );
```

**Pull request merged**